Makefile

```diff
@@ -64,3 +64,13 @@ test: build/xdebug-filter.php
 docs:
 	cd docs && sphinx-build -W -b html -d _build/doctrees . _build/html
 .PHONY: docs
+
+YARN ?= @docker run --rm -it -v ${PWD}:/usr/src/app -w /usr/src/app node:11-alpine yarn
```

Comment:
The node_modules are created as `root`, this is not great… maybe we should add `--user=$(shell id -u)`?

Reply:
yea i deleted `node_modules` and run `make yarn-build` again it's still not the root

Reply:
How do you run the docker daemon? `ps aux|grep dockerd`

Reply:
```bash
silasjoisten     39385   0,0  0,0  4278300    852 s002  S+    2:17pm   0:00.00 grep --color=auto --exclude-dir=.bzr --exclude-dir=CVS --exclude-dir=.git --exclude-dir=.hg --exclude-dir=.svn dockerd

```

Reply:
You're showing me the grep command that found itself… there should be another line for the docker daemon. Maybe the Docker daemon command isn't `dockerd` on your machine?

Reply:
```bash
silasjoisten       596   8,7  4,7 11825908 789848   ??  S    Fr10pm  346:03.57 com.docker.hyperkit -A -u -F vms/0/hyperkit.pid -c 5 -m 7168M -s 0:0,hostbridge -s 31,lpc -s 1:0,virtio-vpnkit,path=vpnkit.eth.sock,uuid=2ef3fbc7-ffe3-47eb-b2aa-4a7c9c6bf7cc -U ce6c7bfc-206a-40e2-b2e8-5c97da0640b9 -s 2:0,ahci-hd,/Users/silasjoisten/Library/Containers/com.docker.docker/Data/vms/0/Docker.raw -s 3,virtio-sock,guest_cid=3,path=vms/0,guest_forwards=2376;1525 -s 4,ahci-cd,/Applications/Docker.app/Contents/Resources/linuxkit/docker-for-mac.iso -s 5,ahci-cd,vms/0/config.iso -s 6,ahci-cd,/Applications/Docker.app/Contents/Resources/linuxkit/docker.iso -s 7,virtio-rnd -l com1,autopty=vms/0/tty,asl -f bootrom,/Applications/Docker.app/Contents/Resources/uefi/UEFI.fd,,
silasjoisten     27501   0,0  0,1 558517232  22388 s001  S+   10:22am   0:01.12 /usr/local/bin/docker exec --interactive --tty dd7bfcc0d52c430c9c826f990285b93a34ff424699c29c676ccdefafd14f02d1 bash
silasjoisten     27498   0,0  0,1  4397488  19968 s001  S+   10:22am   0:00.24 docker-compose exec php-cli bash
silasjoisten     27497   0,0  0,0  4323600   4460 s001  S+   10:22am   0:00.08 docker-compose exec php-cli bash
silasjoisten       589   0,0  0,1 558607496  10588   ??  S    Fr10pm    0:09.26 com.docker.driver.amd64-linux -addr fd:3 -debug
silasjoisten       588   0,0  0,9  4520948 145044   ??  S    Fr10pm    1:16.20 com.docker.vpnkit --ethernet fd:3 --port vpnkit.port.sock --port hyperkit://:62373/./vms/0 --diagnostics fd:4 --pcap fd:5 --vsock-path vms/0/connect --host-names host.docker.internal,docker.for.mac.host.internal,docker.for.mac.localhost --gateway-names gateway.docker.internal,docker.for.mac.gateway.internal,docker.for.mac.http.internal --vm-names docker-for-desktop --listen-backlog 32 --mtu 1500 --allowed-bind-addresses 0.0.0.0 --http /Users/silasjoisten/Library/Group Containers/group.com.docker/http_proxy.json --dhcp /Users/silasjoisten/Library/Group Containers/group.com.docker/dhcp.json --port-max-idle-time 300 --max-connections 2000 --gateway-ip 192.168.65.1 --host-ip 192.168.65.2 --lowest-ip 192.168.65.3 --highest-ip 192.168.65.254 --log-destination asl --udpv4-forwards 123:127.0.0.1:58640 --gc-compact-interval 1800
silasjoisten       587   0,0  7,6  5659532 1269124   ??  S    Fr10pm   28:37.18 com.docker.osxfs serve --address fd:3 --connect vms/0/connect --control fd:4 --log-destination asl
silasjoisten       586   0,0  0,0 558594988   4244   ??  S    Fr10pm    0:02.19 /Applications/Docker.app/Contents/MacOS/com.docker.supervisor -watchdog fd:0
root                77   0,0  0,0  4289832    176   ??  Ss   Fr10pm    0:00.01 /Library/PrivilegedHelperTools/com.docker.vmnetd
silasjoisten     42012   0,0  0,0  4268060    800 s003  R+    3:51pm   0:00.00 grep --color=auto --exclude-dir=.bzr --exclude-dir=CVS --exclude-dir=.git --exclude-dir=.hg --exclude-dir=.svn docker
```

Reply:
Ah you're using MacOS, that's probably why things are that different… can you try adding the piece of code I told you?

Reply:
Yea sure!!

Reply:
I tried this out with the `--user` option. There will be a problem with permissions of cache

Reply:
what problem?

Reply:
The user dont have permissions to access directory of yarn cache

Reply:
Which user? Where is the yarn cache and who does it belong to? Do you mean read access or write access?

Reply:
This is the error message
```bash
error Could not write file "/usr/src/app/yarn-error.log": "EACCES: permission denied, open '/usr/src/app/yarn-error.log'"
error An unexpected error occurred: "EACCES: permission denied, mkdir '/usr/src/app/node_modules'".
```